### ISSUE TITLE
Getting JSON format error when restarting flanneld

### ISSUE THREAD

**Issue by simepo** [coreos/flannel#794]
## Expected Behavior
When starting a new flannel node, I insert config into etcd and start flanneld:

`# etcdctl set /atomic.io/network/config '{"Network": "4.0.0.0/8", "SubnetLen": 24, "Backend": {"Type": "vxlan"}}'`

I expect then to be able to stop and restart flanneld as required.

## Current Behavior
However, if the flanneld is now stopped and restarted, it fails with multiple errors:

`
Aug 09 14:16:46 uklpadsab025a flanneld-start[38968]: E0809 14:16:46.229979   38968 network.go:102] failed to retrieve network config: json: cannot unmarshal string into Go value of type subnet.Config
`

If the config in etcd is updated to include VNI:

`# etcdctl set /atomic.io/network/config '{"Network": "4.0.0.0/8", "SubnetLen": 24, "Backend": {"Type": "vxlan", "VNI": 1}}'`

flanneld now starts without issue.

## Steps to Reproduce (for bugs)

1. Install Flannel
2. Insert config including VNI into etcd
3. Start flanneld fails with JSON parsing errors
4. Update config in etcd without VNI, flanneld starts OK.
4. Restart flanneld - fails.
5. Update config in etcd to include VNI, flanneld restarts OK.

## Context

I am trying to deploy Kubernetes/Docker nodes via Ansible, this issue is making it hard to produce a generalised playbook.

## Your Environment

* Flannel version: flannel-0.7.0-1.el7.x86_64
* Backend used (e.g. vxlan or udp): vxlan
* Etcd version: etcd-3.1.7-1.el7.x86_64
* Kubernetes version (if used): 1.5.1
* Operating System and version: RHEL 7.2

**Comment by tomdee** [comment]
@simepo do you know how that RPM is built or where it comes from? Are you able to check that you have the exact same 0.7.0 flannel binary as is hosted here on github?

Your "steps to repro" and "Current Behaviour" aren't quite the same (if I'm reading it correctly). Do you only hit a problem when you add the VNI to the config?

**Comment by simepo** [comment]
Hi, thanks for responding.

The RPMs all come from Red Hat, they are included in the RHEL 7.2 extras/optional channels.

I guess the summary is:

VNI not in etcd - new node will start / existing node will not start
VNI included in etcd - new node will not start / existing node will start

Hope that helps, please let me know what other relevant details I can provide.

**Comment by tomdee** [comment]
@simepo I can't reproduce this. Can you come up with a simple script (e.g. using
```
docker run --privileged --net=host quay.io/coreos/flannel:v0.9.0
```
and
```
 docker run --rm --net=host quay.io/coreos/etcd etcdctl set /coreos.com/network/config '{ "Network": "10.105.128.0/21","SubnetLen": 26, "Backend": {"Type": "vxlan"}}'
```

That demonstrates the problem?
